### ISSUE TITLE
Inconsistency with DDA rotation angles

### ISSUE THREAD

**Issue by barkls** [manoharan-lab/holopy#201]
DDA rotations do not behave as I would expect according to documentation from [DDA](http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.511.4262&rep=rep1&type=pdf) (top of page 22) and [holopy](http://holopy.readthedocs.io/en/latest/users/concepts.html#rotations-of-scatterers). For a scatterer symmetric around the z-axis (like a cylinder), the first angle should have no effect. The second angle tilts the cylinder into the x-y plane along the x-axis orientation. The third angle rotates the scatterer in the x-y plane. However, this is not what we see:
```
%matplotlib inline
import holopy as hp
from holopy.scattering import Cylinder, calc_holo, DDA
det = hp.detector_grid(shape=100, spacing=0.1)
def show_me_a_hologram(rotation):
    cyl = Cylinder(n=1.59, h=2, d=1, center=(5,7,5), rotation=rotation)
    holo = calc_holo(det, cyl, 1.33, 0.66, (1,0),DDA)
    hp.show(holo)
    
print("we expect these two images to be different but they are the same")
show_me_a_hologram((0,90,0))
show_me_a_hologram((0,90,90))

print("we expect these two images to be the same but they are different")
show_me_a_hologram((0,90,0))
show_me_a_hologram((90,90,0))
```
It seems as if the 1st and 3rd angle arguments are being interchanged. Note that Tmatrix and the ``scatterers.rotated`` method have the expected behaviour. As far as I can tell, this looks like a bug in ADDA itself, not our interfacing with it. Switching the angles on our end could be problematic if ADDA ever fixes the problem.

**Comment by barkls** [comment]
I'm now also questioning the orientation of the hologram images. I would expect rotations of (0, 90, 0) to rotate the cylinder around the y-axis, so that the cylinder lies along the x-axis. However, the images are 
elongated along the y-axis direction.

**Comment by vnmanoharan** [comment]
At the moment I can't run the code because of our usual Windows compilation issues.  Can you attach images of the holograms that are calculated?

**Comment by barkls** [comment]
@vnmanoharan [here is a pdf of my notebook](https://github.com/manoharan-lab/holopy/files/1159176/rotations.pdf)


@anna-wang For the rotation of (0, 90, 0) (first image in each comparison), I expect elongation in the x-direction, not in the y-direction as appears in the images. Note that calling the ``rotated`` method on a ``Spheres``  with the angles (0, 90, 0) rotates around the y-axis (i.e. z-alignment --> x-alignment).

**Comment by anna-wang** [comment]
Hey Solomon - I'm not used to interpreting holograms by eye when they're
this close to the sensor. I think the elongation you're speaking of is
usually 90deg rotated from the actual orientation of the scatterer, because
the tips of the cylinder scatter less than the sides.

Maybe try reconstructing the hologram to check, or alternatively, use

hp.vis.vis2d.show_scatterer_slices

to look at the scatter and its rotations directly.
I'll install A-DDA on my work computer and check too.

**Comment by barkls** [comment]
Anna and I looked into this together last night. It turns out that adda defines the Euler angles differently than we do, so it's not as simple as switching the first and third argument. We rotate a scatterer around the z, y, z axes in the fixed lab frame. adda rotates a scatterer around its own z, y, z axes, where the direction of each axis is affected by the previous rotation. We will need to convert our rotation angles in the lab frame into rotation angles in the scatterer frame before passing to adda.

**Comment by anna-wang** [comment]
Here I'll attach files I just sent Solomon - these are scatterers that
handle rotations before passing to DDA. These are of the shapes I used for
the JQSRT and E.coli and other papers, but have been updated so their
indicators can handle an array of points. You can look at their rotations
with hp.vis.vis2d.show_scatterer_slices.

Also, I just checked and this version of spheroid works with DDA.

**Comment by barkls** [comment]
It turns out the switch to doing rotations in the scatterer reference frame is just a matter of switching the order of alpha and gamma rotations. I've updated the dda code accordingly in an upcoming PR.

**Comment by barkls** [comment]
fixed in #202
